File: genium/src/test/resources/smoke/dates/output/android/com/example/smoke/Dates.java
/*
 *

 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.Date;
public final class Dates extends NativeBase {
    public final static class DateStruct {
        @NonNull
        public Date dateField;
        public DateStruct(@NonNull final Date dateField) {
            this.dateField = dateField;
        }
    }
    /**
     * For internal use only.
     * @exclude
     */
    protected Dates(final long nativeHandle) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native Date dateMethod(@NonNull final Date input);
    @NonNull
    public native Date getDateProperty();
    public native void setDateProperty(@NonNull final Date value);
}

File: gluecodium/src/test/resources/smoke/dates/output/android-kotlin/com/example/smoke/Dates.kt
/*

 *
 */

@file:JvmName("Dates")

package com.example.smoke

import com.example.NativeBase
import java.util.Date

class Dates : NativeBase {

    class DateStruct {
        @JvmField var dateField: Date
        @JvmField var nullableDateField: Date?



        constructor(dateField: Date) {
            this.dateField = dateField
            this.nullableDateField = null
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    external fun dateMethod(input: Date) : Date


    external fun nullableDateMethod(input: Date?) : Date?

    var dateProperty: Date
        external get
        external set

    var dateSet: MutableSet<Date>
        external get
        external set




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

